lib/rucio/core/did_meta_plugins/mongo_meta.py

```diff
@@ -47,7 +55,16 @@ def __init__(self, host=None, port=None, db=None, collection=None):
         if collection is None:
             collection = config.config_get('metadata', 'mongo_collection')
 
-        self.client = pymongo.MongoClient("mongodb://{}:{}/".format(host, port))
+        if user is None:
+            user = config.config_get("metadata", "mongo_user", default=None)
+        if user is not None and password is None:
+            password = config.config_get("metadata", "mongo_password")
+        if user is not None:
+            auth = "{user}:{password}@".format(user=user, password=password)
```

Comment:
this could be `if user is not None and password is not None` to be extra sure that `password` exists

Reply:
actually, it can just be merged with  the if above. If `password is None` and doesn't appears in the configuration, we error.

Reply:
done